src/components/shared/form/controls.tsx

```diff
@@ -0,0 +1,166 @@
+import React, { useState } from 'react';
+import styled from '@styled-components';
+import { red, redLight, white, black } from '../../../styles/constants';
+
+interface FormInputProps {
+  hasError?: boolean;
+}
+
+const FormLabel = styled.label`
+  font-weight: 800;
+  padding: 0.825em 0;
+`;
+const FormInput = styled.input<FormInputProps>`
+  padding: 0.425em;
+  border-radius: 3px;
+  box-shadow: 0 0 1px gray;
+  border: ${(props) =>
+    props.hasError ? `1px solid ${red}` : '1px solid lightgray;'};
```

Comment:
I think we should be using `theme` here instead of the values from `constants.ts`.
You can do something like the following (haven't tested it):
```suggestion
    props.hasError ? `1px solid ${({ theme }) => theme.alert.danger}` : '1px solid lightgray;'};
```

And then consider using the `theme` object elsewhere instead of hardcoded color values